### ISSUE TITLE
Equivalent protocols leads to different results when used as argument types

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5344]
**Describe the bug**
See sample code.

Custom protocol `Iterable` works the same as `typing.Iterable`, which can be confirmed by some two-way assignments. Protocols `Extensible0` and `Extensible1` describes `extend` method (which normally appears in `typing.MutableSequence`). `C0` and `C1` are just simplified versions of `typing.MutableSequence`.

As `Iterable` is equivalent to `typing.Iterable`, `Extensible0` should be equivalent to `Extensible1`, therefore the assignments in the end should all type-check.

It used to work but no longer works with 1.1.314.

**To Reproduce**
Check the sample code with Pyright 1.1,314.

**Expected behavior**
The sample code should type-check as with older versions of Pyright.

**Code or Screenshots**
```python
import typing
from typing import Generic, Protocol, TypeVar

T_co = TypeVar("T_co", covariant=True)
T_contra = TypeVar("T_contra", contravariant=True)

class Iterable(Protocol[T_co]):
    def __iter__(self, /) -> typing.Iterator[T_co]:
        ...

class Extensible0(Protocol[T_contra]):
    def extend(self, iterable: Iterable[T_contra], /) -> None:
        ...

class Extensible1(Protocol[T_contra]):
    def extend(self, iterable: typing.Iterable[T_contra], /) -> None:
        ...

class C0(Generic[T_contra]):
    def extend(self, iterable: Iterable[T_contra], /) -> None:
        raise NotImplementedError

class C1(Generic[T_contra]):
    def extend(self, iterable: typing.Iterable[T_contra], /) -> None:
        raise NotImplementedError

a: Extensible0[str] = C0[str]()
b: Extensible1[str] = C0[str]()
c: Extensible0[str] = C1[str]()
d: Extensible1[str] = C1[str]()
```
![image](https://github.com/microsoft/pyright/assets/15389744/15549fb8-d293-4466-bc08-84d503258e39)

**VS Code extension or command-line**
Pylance v2023.6.21 Pre-Release.
Pyright 1.1.314.

**Additional context**
Mypy Playground: https://mypy-play.net/?mypy=latest&python=3.10&gist=887369ae15ca1e39c11b0569c7c4dbd3

**Comment by erictraut** [comment]
Thanks for reporting the problem. This turns out to be the same issue as https://github.com/microsoft/pyright/issues/5243. A fix is already in place, and it will be included in tomorrow's weekly release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.315, which I just published. It will also be included in this week's insiders release of pylance.
